europeana-image-harvester/index.html

```diff
@@ -0,0 +1,146 @@
+    <script src="https://unpkg.com/contentful-ui-extensions-sdk@3"></script>
+  </head>
+  <body>
+
+    <div>
+      <button id="imageHarvest" class="cf-btn-primary">Harvest</button>
+      <p id="message" style="display: none;"></p>
+    </div>
+
+    <script>
+      'use strict';
+
+      window.contentfulExtension.init(extension => {
+        extension.window.startAutoResizer();
+
+        // show dialog to harvest image + attribution
+        async function showHarvestDialog() {
+          const recordUrl = await extension.dialogs.openPrompt({
+            title: 'Harvest',
+            message: 'Enter a record page URL'
+          });
+          if (!recordUrl) return;
+
+          harvestImageAndAttribution(recordUrl)
+            .then(response => {
+              if (response.valid) {
+                document.getElementById('message').innerText = 'Processing';
+                clearFields();
+                populateFields(response.json, response.recordId);
+              } else {
+                document.getElementById('message').innerText = response.error;
+                document.getElementById('message').style.color = '#d9453f';
+              }
+              document.getElementById('message').style.display = 'block';
+            });
+        }
+
+        // get the image and attribution
+        async function harvestImageAndAttribution (recordUrl) {
+          const recordId = recordUrl.match('\/[0-9]+\/[a-zA-Z0-9_]+$')[0];
+          const apiKey = extension.parameters.installation.apiKey;
+
+          const response = await fetch(`https://api.europeana.eu/api/v2/record${recordId}.json?wskey=${apiKey}`);
+          const responseJson = await response.json();
+          if (response.ok) {
+            return { valid: true, json: responseJson, recordId: recordId };
+          } else {
+            return { valid: false, error: responseJson.error ? responseJson.error : 'unknown' };
+          }
+        }
+
+        // populate fields
+        async function populateFields(response, recordId) {
+          const image = response.object.aggregations[0].edmIsShownBy;
+          if (!image) {
+            document.getElementById('message').innerText = 'No image found';
+            document.getElementById('message').style.color = '#d9453f';
+            return;
+          }
+          
+          // look up image info, attribution (title, provider, creator) and rights statement
+          const providerAggregation = response.object.aggregations[0];
+          const resource = providerAggregation.webResources.find( r => {
+            return r.about === image;
+          });
+
+          const proxies = response.object.proxies;
+          const providerProxy = proxies.find( p => {
+            return p.europeanaProxy === false;
+          });
+
+          const attribution = [languageValue(providerProxy.dcTitle),
+                               languageValue(providerProxy.dcCreator),
```

Comment:
Individual web resources may have their own dc:creator which need to be favoured if present, else fallback to that on the proxy.